addons/optics/CfgWeapons.hpp

```diff
@@ -1,110 +1,51 @@
 class CfgWeapons {
     class ItemCore;
     class InventoryOpticsItem_Base_F;
-    class Default;
-
-    class Binocular: Default {
-        forceOptics = 0;             // Allow using compass with Binocular
-        opticsZoomMin = 0.056889;     // 5.25x power
-        opticsZoomMax = 0.056889;     // 9 px/mil
-        modelOptics = "\z\ace\addons\optics\models\NWD_M22_5x"; // 7 degrees horizontal field of view
-        visionMode[] = {"Normal"}; // Can't use nvgs with binoculars any more than you can with scopes
-        // Fix AI using Binocs on short range - #18737
-        // minRange = 300; // 300 = uses Rangefinder often (runs a few meters, stops, uses RF, repeats)
-        minRange = 500; //500 = seem almost never use it..?
-        minRangeProbab = 0.001;
-        midRange = 1000;
-        midRangeProbab = 0.01;
-        maxRange = 5000;
-        maxRangeProbab = 0.01;
-    };
 
-    // zooming reticle scopes
-    class optic_DMS: ItemCore {
+    // HAMR
+    class optic_Hamr: ItemCore {
         class ItemInfo: InventoryOpticsItem_Base_F {
             class OpticsModes {
-                class Snip;
-                class Iron;
+                class Hamr2Scope;
+                class Hamr2Collimator;
             };
         };
     };
 
-    /*class ACE_optic_DMS: optic_DMS {
+    class ACE_optic_Hamr_2D: optic_Hamr {
         author = ECSTRING(common,ACETeam);
-        _generalMacro = "ACE_optic_DMS";
-        scope = 1;
-        displayName = "LOCALIZE ACE DMS";
-        //descriptionShort = "$STR_A3_CFGWEAPONS_ACC_DMS1";
-        weaponInfoType = "ACE_RscWeaponZeroing";
+        scope = 2;
+        displayName = CSTRING(hamr);
 
-        class ItemInfo: ItemInfo {
-            modelOptics = QPATHTOF(models\ace_shortdot_optics.p3d);
+        class CBA_ScriptedOptic {
+            opticsPPEffects[] = {"ACE_OpticsRadBlur1"};
 
-            class OpticsModes: OpticsModes {
-                class Snip: Snip {
-                    opticsZoomMin = 0.05;
-                    opticsZoomMax = 0.3;
-                    opticsZoomInit = 0.3;
-                    discretefov[] = {};
-                    modelOptics[] = {};
-                };
+            reticleTexture = QPATHTOF(reticles\hamr-reticle65_ca.paa);
+            reticleTextureNight = QPATHTOF(reticles\hamr-reticle65Illum_ca.paa);
+            reticleTextureSize = 1 * 0.9;
```

Comment:
before we set it to a higher zoom than normal hamr (changed `opticsZoomMax`) 
(~2.9x instead of 2x from vanilla)
If we are reverting that, then we also need to change the scale of the reticule or the ranging won't match

![scopes](https://github.com/user-attachments/assets/a2319aa6-49af-4de2-9e28-b944d8c257ac)

this value should be something around `0.53`, 
(can test with `(cba_optics_OpticReticleDetailTextures # 0) set [2, 0.53]` and fire to redraw)
but I'd like to try to find a actual formula for converting old to new
also with the smaller size we'll have to redo the pip hole size

Reply:
`.75 * 6.00094 / 8.59571` = 0.523599 
0.75 from https://github.com/acemod/ACE3/blob/master/addons/optics/CfgRscTitles.hpp#L62
and then the change in FOV/zoom

looks pretty close on the windage marks
![y](https://github.com/user-attachments/assets/850c74e6-95c8-4546-9a82-cc4fbaad4434)

but the problem is this is using `modelOptics = "\x\cba\addons\optics\cba_optic_small_70.p3d";`
and it's still not small enough

I'm not sure, but we might have to add more zooms to cup optics,
or revert the zoom change and go back to the 2.9x?

Reply:
> or revert the zoom change and go back to the 2.9x?

Probably the easier solution, as I have no idea how to create new, smaller CBA optic models.
I imagine I'll have to redo the sizes again and try using different CBA optics. The worry I have is that the PIP optics have 2 models to choose from, which might be limiting.

Reply:
I think just setting all the sizes to `0.75` should work if we keep the zoom the same as before
based on
https://github.com/CBATeam/CBA_A3/blob/master/addons/optics/script_component.hpp#L42-L45
https://github.com/acemod/ACE3/blob/master/addons/optics/CfgRscTitles.hpp#L62-L75

Reply:
> I think just setting all the sizes to `0.75` should work if we keep the zoom the same as before based on https://github.com/CBATeam/CBA_A3/blob/master/addons/optics/script_component.hpp#L42-L45 https://github.com/acemod/ACE3/blob/master/addons/optics/CfgRscTitles.hpp#L62-L75

0.75 works fine for the HAMR, but there is a small gap with Elcan and Pitbull scopes on the top and bottom of the screens. Changing it to 0.8 for both makes the gap disappear, but that means we'd need to change the magnification to something higher.